phpBB/includes/acp/acp_styles.php

```diff
@@ -322,20 +291,24 @@ protected function action_uninstall_confirmed($ids, $delete_files)
 		$uninstalled = array();
 		foreach ($rows as $style)
 		{
-			$result = $this->uninstall_style($style, $delete_files);
-
-			if (is_string($result))
+			// Uninstall style
+			try
 			{
-				$messages[] = $result;
-				continue;
-			}
-			$messages[] = sprintf($this->user->lang['STYLE_UNINSTALLED'], $style['style_name']);
-			$uninstalled[] = $style['style_name'];
+				$this->manager->uninstall($style['style_id']);
+				$uninstalled[] = $style['style_name'];
+				$messages[] = sprintf($this->user->lang['STYLE_UNINSTALLED'], htmlspecialchars($style['style_name']));
 
-			// Attempt to delete files
-			if ($delete_files)
+				// Attempt to delete files
+				if ($delete_files)
+				{
+					$this->manager->delete_style_files($style['style_path']);
+					$messages[] = sprintf($this->user->lang['DELETE_STYLE_FILES_SUCCESS'], htmlspecialchars($style['style_name']));
+				}
+			}
+			catch (exception $e)
 			{
-				$messages[] = sprintf($this->user->lang[$this->delete_style_files($style['style_path']) ? 'DELETE_STYLE_FILES_SUCCESS' : 'DELETE_STYLE_FILES_FAILED'], $style['style_name']);
+				$msg = $this->user->lang($e->getMessage());
+				$messages[] = sprintf($msg, htmlspecialchars($style['style_name']));
```

Comment:
In the filesystem component, when there is an exception, where is the language key translated?